server.go

```diff
@@ -295,19 +299,52 @@ type Server struct {
 	DecorateReader DecorateReader
 	// DecorateWriter is optional, allows customization of the process that writes raw DNS messages.
 	DecorateWriter DecorateWriter
-
+	// UDP and TCP queues
+	queueUDPPacket chan *message
+	queueTCPConn   chan *response
 	// Shutdown handling
 	lock    sync.RWMutex
 	started bool
 }
 
+func (srv *Server) spawnWorkerUDP() {
+	go func() {
```

Comment:
This anonymous `func` should be a named `func` so it shows up in panics and is clearer to debug.

Replace it with something like `func (srv *Server) workerUDP() { for range ... }` and then replace `srv.spawnWorkerUDP()` with `go srv.workerUDP()`.

Reply:
Done